mlflow/openai/utils.py

```diff
@@ -151,6 +151,13 @@ def _validate_model_params(task, model, params):
         )
 
 
+def _exclude_params_from_envs(params, envs):
+    """
+    params passed at inference time should override envs.
+    """
+    return {k: v for k, v in envs.items() if k not in params} if params else envs
```

Comment:
Will this erase env entries if a value submitted for a similar key in params is set to None? Is that intended?

Would a dict unpacking + packing work here to ensure that only valid params values replace the env variable values?

```python
def _exclude_params_from_envs(params, envs):
    """
    params passed at inference time should override envs.
    """
    return {**envs, **(params or {})}
```

Reply:
We shouldn't expect params to set a key as None. If that's the case, it overrides envs value unless we exclude None values from params itself, otherwise `{**envs, **(params or {})}` still overrides the value to None.
`params` is something users explicitly pass at inference time, I think they won't pass it unless they really wants to set the key to None.

Reply:
Yeah that was what I was getting at (checking for None values that would invalidate a config). However, if that's a user's intention, that's their intention. 

For the code suggestion, and my line of thinking.... If a user passes "None" with the code in your PR, it deletes the key instead of preserving the key: None relationship. With dict packing / unpacking, it preserves the state of params regardless of the values supplied by the user. 

Here's an ugly repro:

```python 
# Original function
def _exclude_params_from_envs_original(params, envs):
    """
    params passed at inference time should override envs.
    """
    return {k: v for k, v in envs.items() if k not in params} if params else envs

# Proposed function
def _exclude_params_from_envs_proposed(params, envs):
    """
    params passed at inference time should override envs.
    """
    return {**envs, **(params or {})}

# Test cases
def test_function(func):
    # Test 1: Key in params is set to None and the same key exists in envs
    envs = {"key1": "value1", "key2": "value2"}
    params = {"key1": None}
    result = func(params, envs)
    assert result["key1"] is None, f"Test 1 failed for {func.__name__}!"


    # Test 2: Key in params is set to a non-None value and the same key exists in envs
    params = {"key1": "new_value1"}
    result = func(params, envs)
    assert result["key1"] == "new_value1", f"Test 2 failed for {func.__name__}!"

    # Test 3: Key exists only in params and not in envs
    params = {"key3": "value3"}
    result = func(params, envs)
    assert result["key3"] == "value3", f"Test 3 failed for {func.__name__}!"

    # Test 4: Key exists only in envs and not in params
    params = {}
    result = func(params, envs)
    assert result["key2"] == "value2", f"Test 4 failed for {func.__name__}!"

    print(f"All tests passed for {func.__name__}!")
```

```python 
test_function(_exclude_params_from_envs_proposed)
```
```bash
All tests passed for _exclude_params_from_envs_proposed!
```
```python
test_function(_exclude_params_from_envs_original)
```
```bash
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
/var/folders/cd/n8n0rm2x53l_s0xv_j_xklb00000gp/T/ipykernel_37720/2252776925.py in <cell line: 1>()
----> 1 test_function(_exclude_params_from_envs_original)

/var/folders/cd/n8n0rm2x53l_s0xv_j_xklb00000gp/T/ipykernel_37720/2499167076.py in test_function(func)
     19     params = {"key1": None}
     20     result = func(params, envs)
---> 21     assert result["key1"] is None, f"Test 1 failed for {func.__name__}!"
     22 
     23 

KeyError: 'key1'
```

Is this the behavior you are going for?